src/aiperf/workers/worker.py

```diff
@@ -316,14 +317,16 @@ async def _build_response_record(
     async def _process_response(self, record: RequestRecord) -> Turn | None:
         """Process the response from the inference API call and convert it to a Turn object."""
         resp = self.inference_client.endpoint.extract_response_data(record)
-        # TODO how do we handle reasoning responses in multi turn?
+        # Skip reasoning responses in multi-turn conversations
+        has_reasoning = any(isinstance(r.data, ReasoningResponseData) for r in resp)
+        if has_reasoning:
+            return None
```

Comment:
`ReasoningResponseData` has both reasoning text and content text. so in theory you need to check for the content text field. 

One idea I had was to add a get_content_text() in the base class that just returns get_text() in the base class, but the reasoning class would override it to return just the content and not reasoning.

Or there may be a cleaner way to do it.

However for non-streaming, this is important usecase, because the server will dump the whole reasoning + content in the same response packet.